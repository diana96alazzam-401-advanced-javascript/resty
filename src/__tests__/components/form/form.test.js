import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Form from '../../../components/form/form';

describe('<Form/>', () => {

  it('is alive at application start', () => {
    // const form = shallow(<Form />);
    // expect(form.find('.results .method').exists()).toBeTruthy();
  });
   
  it('is alive at application start', () => {
    // const form = shallow(<Form />);
    // expect(form.find('#GET').text()).toStrictEqual('GET');
    // expect(form.find('#POST').text()).toStrictEqual('POST');
    // expect(form.find('#PUT').text()).toStrictEqual('PUT');
    // expect(form.find('#DELETE').text()).toStrictEqual('DELETE');
  });

  it('properly store the users input into state', () => {
    // const form = mount(<Form />);
    // const getButton = form.find('#POST');
    // const textInput = form.find('#textInput');
    // const formTag = form.find('#form');

    // textInput.simulate('change', { target: { value: 'localhost:3000' } });
    // getButton.simulate('click');
    // formTag.simulate('submit');

    // expect(form.state('request').url).toStrictEqual('localhost:3000');
    // expect(form.state('request').method).toStrictEqual('POST');

  });

  it(' properly clear the state after the form is submitted', () => {
    // const form = mount(<Form />);
    // const formTag = form.find('#form');
    // const getButton = form.find('#POST');
    // const textInput = form.find('#textInput');

    // textInput.simulate('change', { target: { value: 'localhost:3000' } });
    // getButton.simulate('click');
    // formTag.simulate('submit');

    // expect(form.state('url')).toStrictEqual('');
    // expect(form.state('method')).toStrictEqual('');
  });

  it(' properly clear the form after the form is submitted', () => {
    // const form = mount(<Form />);
    // const formTag = form.find('#form');
    // const getButton = form.find('#POST');
    // const textInput = form.find('#textInput');  

    // textInput.simulate('change', {target:{value:'localhost:3000'}});
    // getButton.simulate('click');

    // formTag.simulate('submit');

    // expect(textInput.text()).toStrictEqual('');
    // expect(getButton.hasClass('')).toStrictEqual(true);
  });

  it('render the final result', ()=> {
    // const form = mount(<Form />);
    // const formTag = form.find('#form');
    // const getButton = form.find('#POST');
    // const textInput = form.find('#textInput');  

    // textInput.simulate('change', {target:{value:'localhost:3000'}});
    // getButton.simulate('click');

    // formTag.simulate('submit');

    // expect(form.find('.url').text()).toStrictEqual('localhost:3000');
    // expect(form.find('.method').text()).toStrictEqual('POST');
  })

  it('properly display the users input in the output area on form submit', () => {
    // const rendered = renderer.create(<Form />).toJSON();
    // expect(rendered).toMatchSnapshot();
  });

})
