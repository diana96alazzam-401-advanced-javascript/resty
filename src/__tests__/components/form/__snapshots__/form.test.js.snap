// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form/> properly display the users input in the output area on form submit 1`] = `
<main>
  <form
    onSubmit={[Function]}
  >
    <label>
      <span>
        URL: 
      </span>
      <input
        name="url"
        onChange={[Function]}
        type="text"
      />
      <button
        type="submit"
      >
        GO!
      </button>
    </label>
    <label
      className="methods"
    >
      <span
        className=""
        id="get"
        onClick={[Function]}
      >
        GET
      </span>
      <span
        className=""
        id="post"
        onClick={[Function]}
      >
        POST
      </span>
      <span
        className=""
        id="put"
        onClick={[Function]}
      >
        PUT
      </span>
      <span
        className=""
        id="delete"
        onClick={[Function]}
      >
        DELETE
      </span>
    </label>
  </form>
  <section
    className="results"
  >
    <span
      className="method"
    />
    <span
      className="url"
    />
  </section>
</main>
`;
